Neos.ContentRepositoryRegistry/Classes/Service/CatchUpDeduplicationQueue.php

```diff
@@ -0,0 +1,171 @@
+<?php
+namespace Neos\ContentRepositoryRegistry\Service;
+
+use Neos\Cache\Frontend\FrontendInterface;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\ContentRepository\Core\Projection\ProjectionCatchUpTriggerInterface;
+use Neos\ContentRepository\Core\Projection\Projections;
+
+/**
+ *
+ */
+final readonly class CatchUpDeduplicationQueue
+{
+    public function __construct(
+        private ContentRepositoryId $contentRepositoryId,
+        private FrontendInterface $catchUpLock,
+        private ProjectionCatchUpTriggerInterface $projectionCatchUpTrigger
+    ) {}
+
+    public function requestCatchUp(Projections $projections): void
+    {
+        $queuedProjections = $this->triggerCatchUpAndReturnQueued($projections);
+        $attempts = 0;
+        /** @phpstan-ignore-next-line */
+        while ($queuedProjections->isEmpty() === false) {
+            usleep(random_int(100, 100 + $attempts) + ($attempts * $attempts * 10));
+            $queuedProjections = $this->retryQueued($queuedProjections);
+            $attempts++;
+        }
+    }
+
+    /**
+     * @param class-string $projectionClassName
+     * @return void
+     */
+    public function releaseCatchUpLock(string $projectionClassName): void
+    {
+        $this->setStopped($projectionClassName);
+    }
+
+    private function triggerCatchUpAndReturnQueued(Projections $projections): Projections
+    {
+        $passToCatchUp = [];
```

Comment:
I stumbled upon this name, what about `$projectionsToCatchUp`?

Reply:
right, that makes sense